deploy_windows.ps1

```diff
@@ -1,202 +1,208 @@
-{
-    $version = cmd.exe /c '"npm" --version  2>&1'
-    if ( $version -like "*is not recognized*"){
-        throw System.Management.Automation.CommandNotFoundException
+    # Download the monkey
+    $output = cmd.exe /c "git clone $MONKEY_GIT_URL $monkey_home 2>&1"
+    $binDir = (Join-Path -Path $monkey_home -ChildPath $MONKEY_ISLAND_DIR | Join-Path -ChildPath "\bin")
+    if ( $output -like "*already exists and is not an empty directory.*"){
+        "Assuming you already have the source directory. If not, make sure to set an empty directory as monkey's home directory."
+    } elseif ($output -like "fatal:*"){
+        "Error while cloning monkey from the repository:"
+        $output
+        return
     } else {
-        "Npm already installed"
+        "Monkey cloned from the repository"
+        # Create bin directory
+        New-Item -ItemType directory -path $binDir
+        "Bin directory added"
     }
-}
-catch [System.Management.Automation.CommandNotFoundException]
-{
-    "Downloading npm ..."
-    $webClient.DownloadFile($NPM_URL, $TEMP_NPM_INSTALLER)
-    Start-Process -Wait $TEMP_NPM_INSTALLER
-    Remove-Item $TEMP_NPM_INSTALLER
-}
-#Refresh path 
-$env:Path = [System.Environment]::GetEnvironmentVariable("Path","Machine")
-"Updating npm"
-Push-Location -Path (Join-Path -Path $MONKEY_HOME_DIR -ChildPath $MONKEY_ISLAND_DIR | Join-Path -ChildPath "\cc\ui")
-& npm update
-& npm run dist
-Pop-Location
-
-# Install pywin32
-"Downloading pywin32"
-$webClient.DownloadFile($PYWIN32_URL, $TEMP_PYWIN32_INSTALLER)
-Start-Process -Wait $TEMP_PYWIN32_INSTALLER -ErrorAction Stop
-Remove-Item $TEMP_PYWIN32_INSTALLER
-
-# Create infection_monkey/bin directory if not already present
-$binDir = (Join-Path -Path $MONKEY_HOME_DIR -ChildPath $MONKEY_DIR | Join-Path -ChildPath "\bin")
-New-Item -ItemType directory -path $binaries -ErrorAction SilentlyContinue
-
-# Download upx
-if(!(Test-Path -Path (Join-Path -Path $binDir -ChildPath "upx.exe") )){
-    "Downloading upx ..."
-    $webClient.DownloadFile($UPX_URL, $TEMP_UPX_ZIP)
-    "Unzipping upx"
-    Expand-Archive $TEMP_UPX_ZIP -DestinationPath $binDir -ErrorAction SilentlyContinue
-    Move-Item -Path (Join-Path -Path $binDir -ChildPath $UPX_FOLDER | Join-Path -ChildPath "upx.exe") -Destination $binDir
-    # Remove unnecessary files
-    Remove-Item -Recurse -Force (Join-Path -Path $binDir -ChildPath $UPX_FOLDER)
-    "Removing zip file"
-    Remove-Item $TEMP_UPX_ZIP
-}
 
-# Download mimikatz binaries
-$mk32_path = Join-Path -Path $binDir -ChildPath $MK32_DLL
-if(!(Test-Path -Path $mk32_path )){
-    "Downloading mimikatz 32 binary"
-    $webClient.DownloadFile($MK32_DLL_URL, $mk32_path)
-}
-$mk64_path = Join-Path -Path $binDir -ChildPath $MK64_DLL
-if(!(Test-Path -Path $mk64_path )){
-    "Downloading mimikatz 32 binary"
-    $webClient.DownloadFile($MK64_DLL_URL, $mk64_path)
-}
+    # We check if python is installed
+    try
+    {
+        $version = cmd.exe /c '"python" --version  2>&1'
```

Comment:
You can use `Get-Command` :)

Reply:
You even use it later on

Reply:
I can, but not in this place. Get-Command doesn't output results of a command. What if 'python' command is used for python3 ? In that case we still need to download version 2.7. Also, more readable.